pkg/sql/colexec/execgen/template.go

```diff
@@ -89,50 +89,74 @@ func replaceTemplateVars(
 // and a caller
 //   b(true)
 // this function will generate
-//   x = 3
+//   if true {
+//     x = 3
+//   } else {
+//     x = 4
+//   }
 //   return x
+//
+// but because true is a constant, it will be reduced to
+//
+// x = 3
+// return x
 func monomorphizeTemplate(n dst.Node, info *funcInfo, args []dst.Expr) dst.Node {
+	// Create map from formal param name to arg.
+	paramMap := make(map[string]dst.Expr)
+	for i, p := range info.templateParams {
+		paramMap[p.field.Names[0].Name] = args[i]
+	}
+	n = dstutil.Apply(n, func(cursor *dstutil.Cursor) bool {
+		// Replace all usages of the formal parameter with the template arg.
+		n := cursor.Node()
+		switch n := n.(type) {
+		case *dst.Ident:
+			if arg := paramMap[n.Name]; arg != nil {
+				cursor.Replace(dst.Clone(arg))
+			}
+		}
+		return true
+	}, nil)
+
+	return foldConditionals(n, info, args)
+}
+
+// foldConditionals edits conditional statements to try to remove branches that
+// are statically falsifiable.
+func foldConditionals(n dst.Node, info *funcInfo, args []dst.Expr) dst.Node {
```

Comment:
If the compiler is going to do this for us, it feels like a potential source of errors to try and do it here. I don't know if there's that much of a benefit to the slightly more readable code.